tests/stress/metrics_converter.py

```diff
@@ -0,0 +1,140 @@
+import os
+import sys
+from datetime import datetime, timedelta
+from pathlib import Path
+from dataclasses import dataclass
+import shutil
+
+import pandas as pd
+
+
+@dataclass
+class RunMetrics:
+    start_time: str
+    end_time: str
+    average_time: float
+    total_test_time: timedelta
+    max_value_cpu: float
+    max_value_memory: float
+    average_value_cpu: float
+    average_value_memory: float
+
+
+def get_metrics(
+    csv_metrics_in: Path,
+    run_type: str,
+    run_identifier: str,
+    n_runs: int = 5,
+    metrics_out_folder: Path = Path("metrics"),
+) -> RunMetrics:
+
+    # taking in the fliepath
+    data = pd.read_csv(csv_metrics_in)
+
+    if data.shape[0] == 1:
+        raise IndexError(
+            f"The generated CSV file {csv_metrics_in} appears to be empty. This could be caused by the performance monitor failing to initiate."
+        )
+
+    df2 = data.pivot(index="timeStamp", columns="label", values="elapsed")
+    df2.index = df2.index.map(
+        lambda v: datetime.utcfromtimestamp(v / 1000).strftime("%Y-%m-%d %H:%M:%S.%f")
+    )
+    creation_date = df2.index[0]
+
+    # Remove date information from time value : < 11th char
+    df2.index = df2.index.str.slice(start=11)
+    df2["localhost Memory"] = df2["localhost Memory"] / 1000.0
+    df2["localhost CPU"] = df2["localhost CPU"] / 1000.0
+
+    temp_array = []
+
+    for x in range(0, len(df2.index)):
+        temp = df2.index[x]
+        temp_array.append(temp)
+
+    # print(temp_array)
+    for x in range(1, len(temp_array)):
+        a = datetime.strptime(temp_array[0], "%H:%M:%S.%f")
+        b = datetime.strptime(temp_array[x], "%H:%M:%S.%f")
+        temp_array[x] = (b - a).total_seconds()
+
+    # saving the max values of the CPU and Memmory usage, and the start/end time of the test
+    start_time = df2.index[0]
+    end_time = df2.index[len(df2.index) - 1]
+    average_time = temp_array[len(temp_array) - 1] / n_runs
+    total_test_time = timedelta(seconds=temp_array[len(temp_array) - 1])
+
+    max_value_cpu = round(df2["localhost CPU"].max(), 2)
```

Comment:
Will we have the same values when these tests run on a different env (e.g. on a pipeline or on user's machine - linux/windows)?

Reply:
The stress test results I would expect to be different when run in a different environment + on different hardware. This portion of the ticket had been pushed to outside of scope for generating preliminary results. Perhaps this will form part of ticket #563 
 For the unit tests purpose, we use a prewritten metrics file from which we have calculated the expected results and use this as our comparison.